### PR TITLE
Add clhpp

### PR DESCRIPTION
**Pull request by jakirkham**
Adds a package for clhpp. Includes a C++ interface for OpenCL's API.

cc @inducer

### CODE REVIEW

File: recipes/clhpp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    - pocl >=0.14                # [unix]
+{% endif %}
+
+  run:
+    - khronos-opencl-icd-loader  # [win]
+    - ocl-icd                    # [linux]
+{% if OSX_VARIANT == "pocl" %}
+    - osx-pocl-opencl            # [osx]
+    - pocl >=0.14                # [unix]
+{% endif %}
```
Comment (jakirkham) [inline recipes/clhpp/meta.yaml]: Even though we are just producing C++ headers, they seem to be dependent on the OpenCL implementation used. So have left these in as run time dependencies. That way we know which implementation the C++ headers should be used with.

File: recipes/clhpp/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    - osx_pocl_opencl            # [osx]
+{% endif %}
+
+  always_include_files:
+    - include/CL/cl.hpp
+    - include/CL/cl2.hpp
+
+requirements:
+  build:
+    - toolchain
```
Comment (jakirkham) [inline recipes/clhpp/meta.yaml]: Even though we are not compiling anything, it seems the header generation code likes to check and see what compiler we are going to use. Hence we include the `toolchain`.

File: recipes/clhpp/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+build:
+  number: 0
+  script_env:
+    - OSX_VARIANT
+
+{% if OSX_VARIANT != "native" %}
+  skip: True  # [not (osx and py36)]
+{% else %}
+  skip: True  # [unix and not py36]
+  skip: True  # [win and py<36]
```
Comment (jakirkham) [inline recipes/clhpp/meta.yaml]: Requires C++11. So any Windows compiler that doesn't have support for C++11 is out. As this is not really a Python package, we don't need multiple builds with different Python versions. So Python 3.6 is basically sufficient. If newer versions of Windows compilers come out, we will support them to (as they should have C++11 support).

**Comment by inducer** [comment]
Feel free to add me as maintainer.

File: recipes/clhpp/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  license: MIT-like
+  license_file: LICENSE.txt
+  summary: OpenCL Host API C++ bindings
+
+  dev_url: https://github.com/KhronosGroup/OpenCL-CLHPP
+  doc_url: http://github.khronos.org/OpenCL-CLHPP
+
+extra:
+  recipe-maintainers:
+    - inducer
```
Comment (jakirkham) [inline recipes/clhpp/meta.yaml]: Have added you per your request, @inducer.

File: recipes/clhpp/bld.bat [hunk 5]
```diff
@@ -0,0 +1,17 @@
+copy "%RECIPE_DIR%\LICENSE.txt" "%SRC_DIR%\LICENSE.txt"
```
Comment (jakirkham) [inline recipes/clhpp/bld.bat]: FYI the license should be in the next release. It is already in `master`.

xref: https://github.com/KhronosGroup/OpenCL-CLHPP/pull/21

File: recipes/clhpp/build.sh [hunk 6]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+cp "${RECIPE_DIR}/LICENSE.txt" "${SRC_DIR}/LICENSE.txt"
```
Comment (jakirkham) [inline recipes/clhpp/build.sh]: FYI the license should be in the next release. It is already in `master`.

xref: https://github.com/KhronosGroup/OpenCL-CLHPP/pull/21

File: recipes/clhpp/LICENSE.txt [hunk 7]
```diff
@@ -0,0 +1,25 @@
+Copyright (c) 2008-2015 The Khronos Group Inc.
```
Comment (jakirkham) [inline recipes/clhpp/LICENSE.txt]: FYI the license should be in the next release. It is already in `master`.

xref: https://github.com/KhronosGroup/OpenCL-CLHPP/pull/21

**Comment by jakirkham** [comment]
Anything else needed here?

**Comment by inducer** [comment]
LGTM

**Pull request merged**